lightning/src/chain/onchaintx.rs

```diff
@@ -646,14 +681,28 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 						// outpoints to know if transaction is the original claim or a bumped one issued
 						// by us.
 						let mut set_equality = true;
-						if request.outpoints().len() != tx.input.len() {
-							set_equality = false;
-						} else {
-							for (claim_inp, tx_inp) in request.outpoints().iter().zip(tx.input.iter()) {
-								if **claim_inp != tx_inp.previous_output {
-									set_equality = false;
+						if !request.requires_external_funding() {
```

Comment:
Shouldn't want something more restrictive than `requires_external_funding()` as it's encompassing `HolderFundingOutput` i.e holder's commitment transaction which are not malleable and there is no partial claim of the request outpoints possible ?